### DISCUSSION TITLE
TagInput return null in Livewire Custom Component

### DISCUSSION THREAD

**Discussion by Professor93M** [filamentphp/filament#9554] [Help]
Hello Friends,

I faced an issue with the **TagInput** in the custom Livewire component, it return null if I press Enter or Submit Button

```
public function form(Form $form): Form
{
    return $form
        ->schema([
            Forms\Components\TagsInput::make('skills')
                ->required(),
        ])
        ->statePath('data')
        ->model(Skills::class);
}
```

**Comment by leandrocfe** [comment]
Did you do this?
```php
public ?array $data = [];

public function mount(): void
{
    $this->form->fill();
}

public function submit(): void
{
    $data = $this->form->getState();
    //is the $data['skill'] null here?
}
```

**Reply by Professor93M** [reply]
Yes I did, and the $data return null
if I change it to another input like TextInput it return value without any problem

**Reply by leandrocfe** [reply]
console errors? Could you share the whole code please?

**Comment by Professor93M** [comment]
```
class SkillsForm extends Component implements HasForms
{
    use InteractsWithForms;

    public ?array $data = [];

    public function mount(): void
    {
        $this->form->fill();
    }

    public function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\TagsInput::make('skills')
                     ->required(),
            ])
            ->statePath('data')
            ->model(Skills::class);
    }

    public function create(): void
    {
        $data = $this->form->getState();
        $data['user_id'] = auth()->id();

        $record = Skills::create($data);

        $this->form->model($record)->saveRelationships();
    }

    public function render(): View
    {
        return view('livewire.skills-form');
    }
}
```

Console error: "TypeError: Cannot convert undefined or null to object"

**Reply by leandrocfe** [reply]
Is it a custom page on the admin panel or you are using the form builder?

**Reply by Professor93M** [reply]
It a custom livewire component, I've added a form to a livewire component
like that [Adding a form to a Livewire component](https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component#setting-up-the-livewire-component)

**Reply by leandrocfe** [reply]
Maybe a setup issue? Did you do [these steps](https://filamentphp.com/docs/3.x/forms/installation#existing-laravel-projects)?

**Reply by Professor93M** [reply]
Yes, I think there is a bug with this item, because I checked with the **Edit** it not return any value, but with TextInput the values returned

**Reply by leandrocfe** [reply]
Could you share the project on the github?

**Reply by Professor93M** [reply]
Sorry, I can't.

**Comment by ericthanhtung** [comment]
I also encountered the same situation. form when added works normally. but when updating, the value is not displayed

**Reply by leandrocfe** [reply]
share the whole code please

**Reply by ericthanhtung** [reply]
I cannot publish the project. I can just add you to my project. please give me email

**Reply by leandrocfe** [reply]
Sorry, the code is quite weird; I wasn't able to reproduce it. Please create a repository with minimal code to reproduce this error.

**Comment by ericthanhtung** [comment]
https://github.com/ericthanhtung/example

Product value is not showing when I press update


[image: image.png][image: image.png]

Vào Th 5, 28 thg 12, 2023 vào lúc 19:52 Leandro Costa Ferreira <
***@***.***> đã viết:

**Reply by leandrocfe** [reply]
```php
EditAction::make('update_product')
    ->mountUsing(function (Form $form, LandingPage $record) {
        $form->fill([
            'product' => json_decode($record->product),
        ]);
    })
...
```

**Comment by ericthanhtung** [comment]
Thank you so much

Vào Th 6, 29 thg 12, 2023 vào lúc 04:26 Leandro Costa Ferreira <
***@***.***> đã viết:
